localstack-core/localstack/services/lambda_/provider.py

```diff
@@ -1866,6 +1870,52 @@ def update_alias(
     # =======================================
     # ======= EVENT SOURCE MAPPINGS =========
     # =======================================
+    def check_service_resource_exists(
+        self, service: str, resource_arn: str, function_arn: str, function_role_arn: str
+    ):
+        """
+        Check if the service resource exists and if the function has access to it.
+
+        Raises:
+            InvalidParameterValueException: If the service resource does not exist or the function does not have access to it.
+        """
+        arn = parse_arn(resource_arn)
+        source_client = get_internal_client(
+            arn=resource_arn,
+            role_arn=function_role_arn,
+            service_principal=ServicePrincipal.lambda_,
+            source_arn=function_arn,
+        )
+        if service in ["sqs", "sqs-fifo"]:
+            try:
+                source_client.get_queue_attributes(QueueUrl=arn["resource"])
+            except ClientError as e:
+                if e.response["Error"]["Code"] == "AWS.SimpleQueueService.NonExistentQueue":
+                    raise InvalidParameterValueException(
+                        f"Error occurred while ReceiveMessage. SQS Error Code: {e.response['Error']['Code']}. SQS Error Message: {e.response['Error']['Message']}",
+                        Type="User",
+                    )
+                raise e
+        elif service in ["kinesis"]:
```

Comment:
nit: I missed it before but this could be `elif service == "kinesis"` as I don't think we would check other kind of services that way. Not blocking however! and same for `dynamodb`.

Reply:
oh sorry, I got here late, your team was so quick to merge it 😅